add data processing page (#421)
moj-analytical-services/user-guidance: source/documentation/data-docs/data-faqs/index.md

# Data FAQs

## What types of data are on the Platform?

Data on the Analytical Platform can largely be split into four categories:
- `Raw`: data that has been uploaded to the Analytical Platform without any changes made to it

> - `Curated`: data that has been validated, organised, cleaned and transformed by the Data Engineers ready to be used by Analytical Platform users

Author: My data engineering experience and knowledge is limited, could you write out what you would like here and I will paste in?

> - `Derived`: data that has been transformed, aggegregated, reduce and turned into a data model to fit specific needs of Analytical Platform users

Author: My data engineering experience and knowledge is limited, could you write out what you would like here and I will paste in?

## How do I gain access to existing data?

Access to databases is granted via the [database access repository](https://github.com/moj-analytical-services/data-engineering-database-access). Have a read through the guidance on there. If you are finding the process a little tricky, please ask for help in [#ask-data-engineering](https://moj.enterprise.slack.com/archives/C8X3PP1TN). A data engineer will happily guide you through things.

If you are looking for access to a user created bucket, then the admin of that bucket should be able to grant you access. If you don't know who the admin is, or they are not able to grant you access, then ask in the [#analytical-platform-support](https://app.slack.com/client/T02DYEB3A/C4PF7QAJZ) Slack channel or via [GitHub](https://github.com/ministryofjustice/data-platform-support/issues/new/choose). If the bucket admin is unavailable, the Analytical Platform team will need to receive approval from your line manager before you can be given any access to the bucket.

## Where should I store my own data?

> Data should be stored in an S3 bucket. You can create a new S3 bucket in the [AP Control Panel](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/control-panel.html). Data can be uploaded manually via the AWS console (which can be accessed through the control panel) or you can write it from RStudio, Visual Studio Code or JupyterLab.

Author: Added below the header:

> ⚠️ We can only hold 1,000 S3 buckets on the Analytical Platform. Before you create a new bucket, please check whether you can re-use an existing one, or create a domain-level S3 bucket and use folders and [path specific access](https://user-guidance.analytical-platform.service.justice.gov.uk/data/amazon-s3/#path-specific-access) for project level access.

If your data contains anything that could be considered personal information, you must follow guidance from the data protection team which can be found on the [intranet](https://intranet.justice.gov.uk/guidance/knowledge-information/protecting-information/privacy-reform/).

## How do I read/write data from an s3 bucket?

> **Python**: You can read/write directly from S3 using pandas. However, to get the best representation of the column types in the resulting Pandas dataframe(s), you may wish to use [mojap-arrow-pd-parser](https://github.com/moj-analytical-services/mojap-arrow-pd-parser).

Author: Not sure I quite follow "at some point we will stop allowing people to read files directly". Does this mean people could not read directly from S3? What about unstructured data like images/text files (anything that isn't in an Athena table) surely users still need to be able to access this directly?

> **Python**: You can read/write directly from S3 using pandas. However, to get the best representation of the column types in the resulting Pandas dataframe(s), you may wish to use [mojap-arrow-pd-parser](https://github.com/moj-analytical-services/mojap-arrow-pd-parser).

Author: @SoumayaMauthoorMOJ can we resolve this one now?

# Data FAQs

## What types of data are on the Platform?

Data on the Analytical Platform can largely be split into four categories:
- `Raw`: data that has been uploaded to the Analytical Platform without any changes made to it

> - `Curated`: data that has been validated, organised, cleaned and transformed by the Data Engineers ready to be used by Analytical Platform users

Reviewer: ```suggestion
- `Curated`: data that has been validated, deduplicated and versioned by the Data Engineers ready to be used by Analytical Platform users
```

> - `Derived`: data that has been transformed, aggregated, reduce and turned into a data model to fit specific needs of Analytical Platform users

Reviewer: ```suggestion
- `Derived`: data that has been [denormalized](https://en.wikipedia.org/wiki/Denormalization#:~:text=Denormalization%20is%20a%20strategy%20used,data%20or%20by%20grouping%20data.), aggregated, and turned into a data model to fit specific needs of Analytical Platform users
```

**The Amazon Athena workbench**: If you [log into the AWS console](https://aws.services.analytical-platform.service.justice.gov.uk) and click Services -> Athena, you'll see the Athena workbench. This is good for testing your queries.

**Create a Derived Table**: You can also use [Create a Derived Table](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/create-a-derived-table/index.html) to run SQL statements to query Athena.

If you get `assumed-role/... is not authorized to perform: glue:GetDatabases`, [request database access](https://github.com/moj-analytical-services/data-engineering-database-access#standard-database-access).

**Python**: To run queries and/or read data into a pandas DataFrame, use `pydbtools`. More details are [here](https://github.com/moj-analytical-services/pydbtools). Remember to install the latest version!

> **R**: There is currently no single recommended package for querying databases in R. There is [dbtools](https://github.com/moj-analytical-services/dbtools) which should work on the "old" platform. [Rdbtools](https://github.com/moj-analytical-services/Rdbtools) should work on the "new" platform, but is not officially supported so should be used with caution.

Reviewer: ```suggestion
**R**: There is currently no single recommended package for querying databases in R. There is [dbtools](https://github.com/moj-analytical-services/dbtools) which should work on the "old" platform. [Rdbtools](https://github.com/moj-analytical-services/Rdbtools) should work on the "new" platform. This package is maintained by the analytical platform user community.
```

Reviewer: Not sure what you mean by old and new platform?

Author: That text comes from the existing text on the guidance: https://user-guidance.analytical-platform.service.justice.gov.uk/data/data-faqs/#how-do-i-query-a-database-on-the-platform

I'll update to remove reference to 'old' and 'new'

> ## I am running into memory issues using Python/R, what should I do?

Reviewer: ```suggestion
## What should I use to process my data?

Please refer to [tools for processing my data](#link)

## I am running into memory issues using Python/R, what should I do?
```
